eahub/profiles/receivers.py

```diff
@@ -0,0 +1,88 @@
+import uuid
+from datetime import datetime
+
+import pytz
+from django.core.cache import cache
+from django.db.models.signals import post_save, pre_save
+from django.dispatch import receiver
+
+from eahub.profiles.models import (
+    CauseArea,
+    ExpertiseArea,
+    GivingPledge,
+    OrganisationalAffiliation,
+    Profile,
+    ProfileAnalyticsLog,
+)
+
+profile_fields_to_ignore = ["_state", "_django_cleanup_original_cache"]
+profile_fields_enums_map = {
+    "cause_areas": CauseArea,
+    "giving_pledges": GivingPledge,
+    "expertise_areas": ExpertiseArea,
+    "organisational_affiliations": OrganisationalAffiliation,
+}
+
+
+def save_logs_for_new_profile(instance):
+    action_uuid = uuid.uuid4()
+    time = datetime.utcnow().replace(tzinfo=pytz.utc)
+    for (field, value) in instance.__dict__.items():
```

Comment:
do you think we can use `for field in Profile._meta.get_fields()`? that's the method django is using in their codebase

and then perhaps we can call `getattr(instance, field.name)`

Reply:
This can't handle our enum fields:

        profile = create_profile("test@email.com", "User1")

        profile.name = "User1New"
        profile.cause_areas = [CauseArea.BUILDING_EA_COMMUNITIES]
       profile.save()
>       TypeError: cannot unpack non-iterable ManyToOneRel object